playbooks/roles/load-balancer-v2/templates/haproxy.cfg.ctmpl

```diff
@@ -126,8 +126,16 @@ backend be-{{$config.domain_slug}}
     {{- /* Loop over all active appservers for this instance. */ -}}
     {{- /* Each appserver in this list is a dict of the form {id: int, public_ip: str}. */ -}}
     {{- range $config.active_app_servers }}
+      {{- if .public_ip }}
     server appserver-{{ .id }} {{ .public_ip }}:80 cookie appserver-{{ .id }} {{ if $health_check }}check{{ end }}
+      {{- else}}
+    http-response redirect location / unless { capture.req.uri / }
+    server unscheduled-maintenance-page 127.0.0.1:{{! maintenance_unscheduled_server_port !}}
+      {{- end }}
+    {{- else }}
```

Comment:
This is a `range` - `else`. I'm not familiar with Go's/consul-template's syntax and I didn't find this case in a quick skim of the documentation. I hope it works. Was it tested in practice?

Reply:
Yep, I tested it. It works just the same as Python.

```
for x in []:
  # won't run
else:
  # will run
```

Reply:
That's not very reassuring :-) because the Python one does this:
```
for x in [1]:
  print("will run: for")
else:
  print("will run: else")
```
That is, both run. It's because the `else` runs after the iterations finish in a natural way (without break). There are more differences in error handling but this doesn't affect this template.

In this `range - else` in the template, I guess that in the end we output only 1 `server` directive (either the IP or the preliminary page), never 2. If that's tested, then it's fine.

Reply:
I got the info from [here](https://golang.org/pkg/text/template/):

> **{{range pipeline}} T1 {{else}} T0 {{end}}**
	The value of the pipeline must be an array, slice, map, or channel.
	If the value of the pipeline has length zero, dot is unaffected and
	T0 is executed; otherwise, dot is set to the successive elements
	of the array, slice, or map and T1 is executed.

And indeed, a look at `/etc/haproxy/haproxy.cfg` shows that all the backends have a single server line (except the backends with multiple appservers -- which have one per appserver).